bindings/jupyter/js/src/widgets/index.js

```diff
@@ -1,15 +1,35 @@
+const Jupyter = require('base/js/namespace');
 const widgetBuilder = require('./react-widget-builder').default;
+const req = require('../../../jrequirements.json');
+
+// TODO: Move these to jrequirements.json
+const reqExt = {
+  Manifold: function() {
+    this.listenTo(
+      this.model,
+      'change:segments',
```

Comment:
Could you elaborate on how this `reqExt` work in the PR description?

From line 31 it seems it's a map from the component name (e.g. Manifold) to a callback function needed for `widgetBuilder`.

The callback function itself contains one or many (other than `change:segments` I'd assume) actionId (line 10) to action logic (line 11-25) mappings for how Jupyter should behavior when certain interactions happen inside the app.

Is this the correct understanding?

Reply:
https://github.com/uber/manifold/issues/46 is updated by @kenns29, for reference.